audit_log/tests/test_audit_logging.py

```diff
@@ -0,0 +1,239 @@
+from datetime import datetime, timezone
+
+import pytest
+from django.contrib.auth.models import AnonymousUser
+from freezegun import freeze_time
+
+from audit_log import audit_logging
+from audit_log.enums import Operation, Status
+from audit_log.models import AuditLogEntry
+
+_common_fields = {
+    "audit_event": {
+        "origin": "atv",
+        "status": "SUCCESS",
+        "date_time_epoch": 1590969600000,
+        "date_time": "2020-06-01T00:00:00.000Z",
+        "actor": {
+            "role": "OWNER",
+            "user_id": "",
+            "ip_address": "192.168.1.1",
+            "provider": "",
+        },
+        "operation": "READ",
+        "target": {
+            "id": "",
+            "type": "User",
+        },
+        "additional_information": "",
+    }
+}
+
+
+def test_log_origin(user):
+    audit_logging.log(user, "", Operation.READ, user)
+
+    message = AuditLogEntry.objects.first().message
+    assert message["audit_event"]["origin"] == "atv"
+
+
+@freeze_time("2020-06-01T00:00:00Z")
+def test_log_current_timestamp(user):
+    logging_datetime = datetime.now(tz=timezone.utc)
+    audit_logging.log(user, "", Operation.READ, user)
+
+    message = AuditLogEntry.objects.first().message
+    logged_date_from_date_time_epoch = datetime.fromtimestamp(
+        int(message["audit_event"]["date_time_epoch"]) / 1000, tz=timezone.utc
+    )
+    assert logging_datetime == logged_date_from_date_time_epoch
+    logged_date_from_date_time = datetime.strptime(
+        message["audit_event"]["date_time"], "%Y-%m-%dT%H:%M:%S.%f%z"
+    )
+    assert logging_datetime == logged_date_from_date_time
+
+
+@pytest.mark.parametrize("operation", list(Operation))
+def test_log_owner_operation(fixed_datetime, user, operation):
+    audit_logging.log(
+        user, "", operation, user, get_time=fixed_datetime, ip_address="192.168.1.1"
+    )
+
+    message = AuditLogEntry.objects.first().message
+    assert message == {
+        **_common_fields,
+        "audit_event": {
+            **_common_fields["audit_event"],
+            "operation": operation.value,
+            "actor": {
+                "role": "OWNER",
+                "user_id": str(user.uuid),
+                "ip_address": "192.168.1.1",
+                "provider": "",
+            },
+            "target": {
+                "id": str(user.uuid),
+                "type": "User",
+            },
+        },
+    }
```

Comment:
Would it maybe make sense to use [`snapshots`](https://pypi.org/project/snapshottest/) here as well? (for all this big dict values)

Reply:
If we need to test very specific things like the `actor.role` or smt like that, the we can just test that and then match the snapshot.

```suggestion
    assert message.get("actor", {}).get("role") == "OWNER"
    snapshot.assert_match(message)
```

Reply:
Using snapshots sounds good. :+1:

Reply:
Except that user `uuid` does change. :thinking:

Reply:
Resolved